pkg/types/ccipocr3/generic_types.go

```diff
@@ -82,51 +82,85 @@ func (c ChainSelector) String() string {
 	return fmt.Sprintf("ChainSelector(%d)", c)
 }
 
-type CCIPMsg struct {
-	CCIPMsgBaseDetails
-	ChainFeeLimit   BigInt        `json:"chainFeeLimit"`
-	Nonce           uint64        `json:"nonce"`
-	Sender          types.Account `json:"sender"`
-	Receiver        types.Account `json:"receiver"`
-	Strict          bool          `json:"strict"`
-	FeeToken        types.Account `json:"feeToken"`
-	FeeTokenAmount  BigInt        `json:"feeTokenAmount"`
-	Data            []byte        `json:"data"`
-	TokenAmounts    []TokenAmount `json:"tokenAmounts"`
-	SourceTokenData [][]byte      `json:"sourceTokenData"`
-	// Metadata is used as a backup for any additional data that is not covered by the fields above.
-	Metadata CCIPMsgMetadata `json:"metadata"`
-}
-
-type CCIPMsgMetadata struct {
-	// Version of the message metadata. Required in order to be able to parse the metadata
-	// by the underlying implementation.
-	Version string `json:"version"`
-	// Data is the metadata payload. The underlying implementation should know how to parse this data.
-	Data []byte `json:"data"`
-}
-
-type TokenAmount struct {
-	Token  types.Account
-	Amount *big.Int
-}
-
-func (c CCIPMsg) String() string {
+// Message is the generic Any2Any message type for CCIP messages.
+// It represents, in particular, a message emitted by a CCIP onramp.
+// The message is expected to be consumed by a CCIP offramp after
+// translating it into the appropriate format for the destination chain.
+type Message struct {
+	// Header is the family-agnostic header for OnRamp and OffRamp messages.
+	// This is always set on all CCIP messages.
+	Header RampMessageHeader `json:"header"`
```

Comment:
Is the `Header.Hash` field populated on OnRamp or it's computed offChain/offRamp as it was the case so far?

Reply:
I think like before, `Hash` would be populated by the plugin. `MessageID` however is computed onchain by the onramp.